### PR TITLE
Move release api structs to main module

### PR DESCRIPTION
**Pull request by g-gaston**
*Description of changes:*
Now the main eks-a module doesn't depend on the release cli module since the api structs are part of itself. The release cli code is its own module that now depends on the main eks-a module. This should facilitate dependency updates in the main module, specially by dependabot.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: release/cli/go.mod [hunk 1]
```diff
@@ -1,10 +1,11 @@
-module github.com/aws/eks-anywhere/release
+module github.com/aws/eks-anywhere/release/cli
 
 go 1.20
 
 require (
-	github.com/aws/aws-sdk-go v1.38.40
+	github.com/aws/aws-sdk-go v1.42.23
 	github.com/aws/aws-sdk-go-v2 v1.22.1
+	github.com/aws/eks-anywhere v1.18.0
```
[the remark this answers is not in the document]
Comment (g-gaston) [reply release/cli/go.mod:8]: oh weird, let me check

**Comment by chrisdoherty4** [comment]
I appreciate there's some sort of Go module issue here but I'm not clear on the new setup. My understanding is as follows:

* `/release` is no longer a Go module.
* `/release/cli` is a module that contains code _to build a release CLI only_.
* Anything left in `/release` (not part of the `cli` module) is required by EKS-A proper, hence is no part of the EKS-A module.
* End result is no cyclic module dependencies.

File: release/cli/go.mod (hunk 1 above)
[the remark this answers is not in the document]
Comment (g-gaston) [reply release/cli/go.mod]: fixed

**Comment by g-gaston** [comment]
> I appreciate there's some sort of Go module issue here but I'm not clear on the new setup. My understanding is as follows, could you confirm/correct:
> 
>     * `/release` is no longer a Go module.
> 
>     * `/release/cli` is a module that contains code _to build a release CLI only_.
> 
>     * Anything left in `/release` (not part of the `cli` module) is required by EKS-A proper, hence is no part of the EKS-A module.
> 
>     * End result is no cyclic module dependencies.

Yeah, all correct except the third point

Anything left in `/release` (not part of the `cli` module) is required by EKS-A proper, hence **is part** of the EKS-A module.

**Review by chrisdoherty4 — APPROVED** [review]

**Pull request merged**